src/_pytest/terminal.py

```diff
@@ -956,7 +956,28 @@ def show_skipped(lines):
 
 def _get_pos(config, rep):
     nodeid = config.cwd_relative_nodeid(rep.nodeid)
-    return nodeid
+    path, _, testname = nodeid.partition("::")
+    if not testname:
+        return nodeid
+
+    # Append location (line number).
+    # This uses the first traceback entry for the location in the test itself
+    # (rather than reprcrash, which might be less relevant for going to
+    # directly, e.g. pexpect failures in pytest itself).
+    try:
+        testloc = rep.longrepr.reprtraceback.reprentries[0].reprfileloc
+    except AttributeError:
+        return nodeid
+
+    if isinstance(testloc.path, six.string_types):
```

Comment:
I think you can do just this:

```python
testloc_path = py.path.local(testloc.path)
```

If `testloc.path` is already a `py.path.local` instance, then it should be a noop (or very cheap).

Reply:
Yeah, this was more to see if it is needed after all (via coverage).
IIRC I've seen some inconsistency when testing, but only one case is used IIRC^2.